src/main/java/io/jenkins/plugins/gitlabbranchsource/helpers/GitLabPipelineStatusNotifier.java

```diff
@@ -276,11 +302,22 @@ private static void sendNotifications(Run<?, ?> build, TaskListener listener) {
         try {
             GitLabApi gitLabApi = GitLabHelper.apiBuilder(source.getServerName());
             LOGGER.log(Level.FINE, String.format("Notifiying commit: %s", hash));
-            gitLabApi.getCommitsApi().addCommitStatus(
```

Comment:
The block could be rewritten in this way:
```code
            Integer projectId = (revision instanceof MergeRequestSCMRevision) ? getSourceProjectId(build.getParent(), gitLabApi, source.getProjectPath()) : source.getProjectPath()
            gitLabApi.getCommitsApi().addCommitStatus(
                projectId,
                hash,
                state,
                status);
```

Reply:
Thanks for the tip. That didn't quite work, it produced the below error.

```
[ERROR] /build/src/main/java/io/jenkins/plugins/gitlabbranchsource/helpers/GitLabPipelineStatusNotifier.java:[306,178] incompatible types: bad type in conditional expression
```

I did expand upon your suggested change and managed to get it working. PR updated.

Reply:
Nevermind, reverted. Had some linting errors and I can't really be bothered spending any more time on this.